batchcloser/closers.go

```diff
@@ -84,12 +88,7 @@ type BatchCloser struct {
 func (bc *BatchCloser) Close() error {
 	var errs errorsbp.Batch
 	for _, closer := range bc.closers {
-		if err := closer.Close(); err != nil {
-			errs.Add(CloseError{
-				Cause:  err,
-				Closer: closer,
-			})
-		}
+		errs.AddPrefix(fmt.Sprintf("%#v", closer), closer.Close())
```

Comment:
🔕  I kinda feel like this is a little too magic? It still makes sense to me to check:

```go
if err := closer.Close(); err != nil {
}
```

I know this way works, but it's not clear that the errors are only added if `err` is non-nil from this part of the code.  I don't feel super strongly though if you prefer this way.

Reply:
I think we should discourage users from doing the extra nil check and leave that to `Batch`. This leads to simpler code, and Baseplate.go code should serve as an example :)

Also that behavior is documented in `Batch`, so I disagree that it's "too magic" :)

Reply:
Oh wait I forgot to document that part. Let me add it real quick 😬